web/pandas/pdeps/0013-standardize-apply.md

```diff
@@ -0,0 +1,333 @@
+
+```python
+>>> small_ser.apply({"sqrt" :"sqrt", "abs" : np.abs})
+sqrt  0    1.000000
+      1    1.414214
+      2    1.732051
+abs   0    1.000000
+      1    2.000000
+      2    3.000000
+dtype: float64
+```
+
+These two should give same-shaped output for consistency. Using `Series.transform` instead of `Series.apply`, it returns a `DataFrame` in both cases and I think the dictlike example above should return a `DataFrame` similar to the listlike example.
+
+Minor additional info: listlikes and dictlikes of aggregation arguments do behave the same, so this is only a problem with dictlikes of transforming arguments when using `apply`.
+
+This PDEP proposes that the result from giving list-likes and dict-likes to `Series.apply` will have the same shape as when given list-likes currently:
+
+```python
+>>> small_ser.apply(["sqrt", np.abs])
+       sqrt  absolute
+0  1.000000         1
+1  1.414214         2
+2  1.732051         3
+>>> small_ser.apply({"sqrt" :"sqrt", "abs" : np.abs})
+       sqrt  absolute
+0  1.000000         1
+1  1.414214         2
+2  1.732051         3
+```
+
+## Proposal
+
+With the above in mind, it is proposed that:
+
+1. When given a callable, `Series.apply` always operate on the series. I.e. let `series.apply(func)` be similar to `func(series)` + the needed additional functionality.
+2. When given a list-like or dict-like, `Series.apply` will apply each element of the list-like/dict-like to the series. I.e. `series.apply(func_list)` will be similar to `[series.apply(func) for func in func_list]` + the needed additional functionality
+3. The changes made to `Series.apply`will propagate to `Series.agg` and `Series.transform` as needed.
+
+The difference between `Series.apply()` & `Series.map()` will then be that:
+
+* `Series.apply()` makes the passed-in callable operate on the series, similarly to how `(DataFrame|SeriesGroupby|DataFrameGroupBy).apply` operate on series. This is very fast and can do almost anything,
+* `Series.map()` makes the passed-in callable operate on each series data elements individually. This is very flexible, but can be very slow, so should only be used if `Series.apply` can't do it.
+
+so, this API change will help make Pandas `Series.(apply|map)` API  clearer without losing functionality and let their functionality  be explainable in a simple manner, which would be a win for Pandas.
+
+The result from the above change will be that `Series.apply` will operate similar to how `DataFrame.apply` works already per column, similar to how `Series.map` operates similar to how `DataFrame.map` works per column. This will give better coherence between same-named methods on `DataFrames` and `Series`.
+
+## Deprecation process
+
+To change the behavior to the current behavior will have to be deprecated. This can be done by adding a `by_row` parameter to `Series.apply`, which means, when `by_rows=False`, that `Series.apply` will not operate elementwise but Series-wise.
```

Comment:
The question is what would be the default behavior if `by_row` is not specified?

Reply:
That will emit a deprecation warning. I've made text change to clarify this.